ADPhantomApp/src/ADPhantom.cpp

```diff
@@ -1451,6 +1477,128 @@ void ADPhantom::phantomCameraTask()
+        rangeValid=false;
+        setStringParam(ADStatusMessage, "end_cine value invalid");
+      } else if(uni_frame_lim){ //Start frame and end frame are applied to every cine
+        int first_frame = 0;
+        int last_frame = 0;
+        for(int cine{start_cine}; cine <= end_cine; cine++){
+          debug(functionName, "Starting sanity checks on cine ", cine);
+          getIntegerParam(PHANTOM_CnFirstFrame_[cine], &first_frame);
+          getIntegerParam(PHANTOM_CnLastFrame_[cine], &last_frame);
+          if(start_frame < first_frame || start_frame > last_frame){
+            rangeValid=false;
+            char message[256];
+            sprintf(message, "start_frame value invalid in cine %d", cine);
+            setStringParam(ADStatusMessage, message);
+            break;
+          } else if(end_frame < first_frame || end_frame > last_frame ){
+            rangeValid=false;
+            char message[256];
+            sprintf(message, "end_frame value invalid in cine %d", cine);
+            setStringParam(ADStatusMessage, message);
+            break;
+          } else if(end_frame < start_frame){
+            rangeValid=false;
+            setStringParam(ADStatusMessage, "start_frame cannot be after end_frame");
+            break;
+          }
+        }
+      } else{ //Start frame refers to start cine and end frame refers to end cine
+        int start_cine_first_frame = 0;
+        int start_cine_last_frame = 0;
+        int end_cine_first_frame = 0;
+        int end_cine_last_frame = 0;
+        getIntegerParam(PHANTOM_CnFirstFrame_[start_cine], &start_cine_first_frame);
+        getIntegerParam(PHANTOM_CnLastFrame_[start_cine], &start_cine_last_frame);
+        getIntegerParam(PHANTOM_CnFirstFrame_[end_cine], &end_cine_first_frame);
+        getIntegerParam(PHANTOM_CnLastFrame_[end_cine], &end_cine_last_frame);
+
+        if (start_frame < start_cine_first_frame || start_frame > start_cine_last_frame) {
+          rangeValid=false;
+          setStringParam(ADStatusMessage, "start_frame value invalid");
+        } else if(end_frame < end_cine_first_frame || end_frame > end_cine_last_frame) {
+          rangeValid=false;
+          setStringParam(ADStatusMessage, "end_frame value invalid");
+        } else if (start_cine == end_cine && end_frame < start_frame) {
+          rangeValid=false;
+          setStringParam(ADStatusMessage, "end_frame cannot be less than start_frame within a cine");
+        }
+      }
+
+      if(rangeValid){
+        // Attach to the correct port
+        //status = attachToPort("dataPort");
+
+        // Download the timestamp information
+        status = readoutTimestamps(start_cine, end_cine, start_frame, end_frame, uni_frame_lim);
+
+        // Download the data and process arrays
+        status = readoutDataStream(start_cine, end_cine, start_frame, end_frame, uni_frame_lim);
+
+        setStringParam(ADStatusMessage, "Ready");
+      }
+   }  
```

Comment:
If the rangeValid flag is false is it worth setting the ADStatus to error as well as setting the ADStatusMessage?